bot/code_review_bot/revisions.py

```diff
@@ -222,45 +218,50 @@ def from_decision_task(task: dict, phabricator: PhabricatorAPI):
         response.raise_for_status()
         description = response.json()["desc"]
         match = REGEX_PHABRICATOR_COMMIT.search(description)
-        if match is not None:
-            url, revision_id = match.groups()
-            revision_id = int(revision_id)
-            logger.info("Found phabricator revision", id=revision_id, url=url)
-        else:
+        if match is None:
             raise Exception(f"No phabricator revision found in commit {commit_url}")
 
+        url, revision_id = match.groups()
+        revision_id = int(revision_id)
+        logger.info("Found phabricator revision", id=revision_id, url=url)
+
         # Lookup the Phabricator revision to get details (phid, title, bugzilla_id, ...)
         revision = phabricator.load_revision(rev_id=revision_id)
 
-        # Search the Phabricator diff with same commit identifier
-        diffs = phabricator.search_diffs(
-            revision_phid=revision["phid"], attachments={"commits": True}
-        )
-        diff = next(
-            iter(
-                d
-                for d in diffs
-                if d["attachments"]["commits"]["commits"][0]["identifier"]
-                == mercurial_revision
-            ),
-            None,
-        )
-        assert (
-            diff is not None
-        ), f"No Phabricator diff found for D{revision_id} and mercurial revision {mercurial_revision}"
-        logger.info("Found phabricator diff", id=diff["id"])
+        diff_attrs = {}
+        if task["payload"]["env"]["GECKO_HEAD_REPOSITORY"] != REPO_MOZILLA_CENTRAL:
+            # Search the Phabricator diff with the same commit identifier, except on Mozilla-central
+            diffs = phabricator.search_diffs(
+                revision_phid=revision["phid"], attachments={"commits": True}
+            )
+            diff = next(
+                iter(
+                    d
+                    for d in diffs
+                    if d["attachments"]["commits"]["commits"][0]["identifier"]
+                    == mercurial_revision
+                ),
+                None,
+            )
+            assert (
+                diff is not None
+            ), f"No Phabricator diff found for D{revision_id} and mercurial revision {mercurial_revision}"
+            logger.info("Found phabricator diff", id=diff["id"])
+            diff_attrs = {
+                "diff": diff["id"],
+                "diff_id": diff["id"],
```

Comment:
WHat's the use of `Revision.diff_id` if it's the same value as the pre-existing `diff` ?

You can simply introduce `diff_phid`

Reply:
You are right, `diff` should store the complete diff dictionary and not just the ID (in order to keep the current behavior)